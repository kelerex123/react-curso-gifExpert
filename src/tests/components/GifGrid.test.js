import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');


describe('Prueba con <GifGrid />', () => {


    const catategory = "One Punch"
    //let wrapper = shallow(<GifGrid category={catategory} key={catategory}/>)

    // beforeEach( () => {
    //     //jest.clearAllMocks();
    //     wrapper = shallow(<GifGrid category={catategory} key={catategory}/>)
    // })

    test('debe hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
                data:[],
                loading:true
        })

        const wrapper = shallow(<GifGrid category={catategory} key={catategory}/>)
        expect(wrapper).toMatchSnapshot();
        
    })

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cualquier/cosa.jpg',
                title: 'Cualquier cosa'
            },
            {
                id: 'AEA',
                url: 'https://localhost/cualquier/cosa.jpg',
                title: 'Cualquier cosa'
            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={catategory} key={catategory}/>)
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })

})