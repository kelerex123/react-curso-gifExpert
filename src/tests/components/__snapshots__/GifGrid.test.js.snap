// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con <GifGrid /> debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
